lib/archethic/contracts/worker.ex

```diff
@@ -162,8 +172,16 @@ defmodule Archethic.Contracts.Worker do
     Logger.debug("Contract execution started", meta)
 
     with true <- enough_funds?(contract_tx.address),
+         {:ok, calls} <- TransactionChain.fetch_contract_calls(contract_tx.address),
          {:ok, next_tx = %Transaction{}} <-
-           Interpreter.execute(:oracle, contract, oracle_tx, skip_inherit_check?: true),
+           Interpreter.execute(
+             :oracle,
+             contract,
+             oracle_tx,
+             # we append the oracle_tx to the calls in case it is missing due to race condition
+             Enum.uniq([oracle_tx | calls]),
```

Comment:
I don't think this is a valid statement.
Oracle transaction does not target explicitly smart contract, hence it should not be part of the calls . It's for the reason there is a distinction between trigger and call transactions.